### DISCUSSION TITLE
What's the reason to have infinite inertia by default for KinematicBodies? Why is inertia not based on velocity?

### DISCUSSION THREAD

**Discussion by golddotasksquestions** [godotengine/godot-proposals#4001] [Engine Core]
Users generally seem to assume KinematicBodies to work with RigidBordies by default without them pushing RigidBodies through other RigidBodies, walls and floors. I mean ... so would I. 

On community channels, almost like clockwork I periodically share this video on the subject (by KidsCanCode):
https://www.youtube.com/watch?v=C-Sn55e5wnk

[Latest example from today](https://www.reddit.com/r/godot/comments/svospj/is_there_a_way_to_avoid_collision_between_just/hxhmbhw/?context=3).

Why not have the default KinematicBody inertia inferred by the it's velocity for example?

**Comment by MewPurPur** [comment]
Agreed, it should be false by default. Sure it breaks compatibility between 3.x releases, but there's probably like 8 people who are actually utilizing infinite intertia. Moreover, in 4.0 this has been removed and CharacterBodies cannot move RigidBodies, so it would ease migration too.

**Comment by golddotasksquestions** [comment]
Hm, I'm really not sure not for the KinematicBody (CharacterBody in Godot4), not being able to push RigidBodies at all by default is any better. I already see the same post on community forms everywhere: "Why can't I push the this RigidBody box with my CharacterBody?"

I think the expectation is that KinematicBodies/CharacterBodies can push RigidBodies, however not through or into other (Static)bodies.
That's way I am asking why not link the inertia by default to movement?

**Reply by MewPurPur** [reply]
It's better as a default, because instead of trying to address glitchy interactions and thinking you've done things wrong, you can directly approach the question as "How to push RigidBody with a KinematicBody".

Maybe we can have something like a "mass" for kinematic bodies or something to determine how they interact with RigidBodies. But that would still be the best with 0 as default.

**Comment by eon-s** [comment]
Move methods are just "hacks" to make so kinematics can detect bodies and make them usable as characters easily. In general, these methods are made so they try to solve an overlap and restore the kinematic (who does not care about the world like pure statics) to a non-overlaping state.

Making them move and ignore overlap to particular bodies is an exception.

BTW, this also happens because collisions are solved in pairs until Godot 3.x, in Godot 4 masks and layers work one way so the new bodies using move ALWAYS stop on collision, but will ignore the bodies that detect them letting them push undetected bodies like dynamic rigids.

**Reply by wareya** [reply]
> In general, these methods are made so they try to solve an overlap and restore the kinematic (who does not care about the world like pure statics) to a non-overlaping state.

They actually go out of their way to avoid ever entering an overlapping state; it's just that the first step of this process involves finding an open space if the kinematic body is indeed overlapping something when you start moving it. This is part of why they can only have zero or infinite inertia.
